src/include/fin.hpp

```diff
@@ -129,6 +129,67 @@ class BaseFin
         return 0;
     }
 
+    bool LoadJsonKernelList(const miopen::Handle& handle,
+                            const json kernel_objects,
+                            json res_item)
+    {
+        for(const auto& kernel_obj : kernel_objects)
+        {
+            const auto size          = kernel_obj["uncompressed_size"];
+            const auto md5_sum       = kernel_obj["md5_sum"];
+            const auto encoded_hsaco = kernel_obj["blob"];
+            const auto decoded_hsaco = base64_decode(encoded_hsaco);
+            std::vector<char> hsaco;
+            try
+            {
+                hsaco = miopen::decompress(decoded_hsaco, size);
+            }
+            catch(const std::exception& e)
+            {
+                std::cerr << "Binary decompression failed, will try re-compiling: " << e.what()
```

Comment:
You are not exactly recompiling the same k object. Did you mean to say something like 'Skipping' instead?

Reply:
Changed to skipping.